samples/Azure.ResourceManager.Sample/Generated/Extensions/SubscriptionResourceExtensionClient.cs

```diff
@@ -494,20 +494,18 @@ public virtual Response<VirtualMachineImage> GetVirtualMachineImage(AzureLocatio
         /// <param name="publisherName"> A valid image publisher. </param>
         /// <param name="offer"> A valid image publisher offer. </param>
         /// <param name="skus"> A valid image SKU. </param>
-        /// <param name="expand"> The expand expression to apply on the operation. </param>
-        /// <param name="top"> The Integer to use. </param>
-        /// <param name="orderby"> The String to use. </param>
+        /// <param name="options"> A class representing the optional parameters in GetVirtualMachineImages method. </param>
         /// <param name="cancellationToken"> The cancellation token to use. </param>
         /// <returns> An async collection of <see cref="VirtualMachineImageResource" /> that may take multiple service requests to iterate over. </returns>
-        public virtual AsyncPageable<VirtualMachineImageResource> GetVirtualMachineImagesAsync(AzureLocation location, string publisherName, string offer, string skus, string expand = null, int? top = null, string orderby = null, CancellationToken cancellationToken = default)
+        public virtual AsyncPageable<VirtualMachineImageResource> GetVirtualMachineImagesAsync(AzureLocation location, string publisherName, string offer, string skus, SampleGetVirtualMachineImagesOptions options, CancellationToken cancellationToken = default)
```

Comment:
Do we reviewed the name of this option bag? This reads weird. It would be better if we change this to `SampleVirtualMachineImagesGetOptions`

Reply:
No, we haven't reviewed the name. Currently I choose this [proposal](https://github.com/Azure/autorest.csharp/issues/1630#issuecomment-1018265506) to define the name for the option bag: {Resource Name/Collection Name/RP Namespce}{Method Name}Options. From the perspective of ensuring option bag naming format consistency, I didn't make any special treatment. But maybe we should reconsider because your suggestion looks better.

Reply:
Ah, I see. This is weird because it is in the extension class.
I believe the pattern makes sense, and the result here is fine.

Please note that this pattern might create duplicate type names. For instance, it is very likely that we have methods with the same names both in resource and extension, and there is a chance that the resource has the same name as the RP, if these two things happen, we will have the same options bag name on two different operations.

Also we might need to make sure the methods that comes from the same operation uses the same option bag type - for instance, the resource get operation is generated into multiple places (resource, collection/extension)